bfs.go

```diff
@@ -56,3 +59,36 @@ type Iterator interface {
 type supportsCopying interface {
 	Copy(context.Context, string, string) error
 }
+
+// --------------------------------------------------------------------
+
+// Resolve opens a bucket from a URL string.
+func Resolve(ctx context.Context, u *url.URL) (Bucket, error) {
+	registryLock.Lock()
+	resv, ok := registry[u.Scheme]
+	registryLock.Unlock()
+	if !ok {
+		return nil, fmt.Errorf("bfs: unkown URL scheme %q", u.Scheme)
+	}
+
+	return resv(ctx, u)
+}
+
+// Resolver constructs a bucket from a URL.
+type Resolver func(context.Context, *url.URL) (Bucket, error)
+
+var (
+	registry     = map[string]Resolver{}
+	registryLock sync.Mutex
+)
+
+// RegisterProtocol registers a new protocol with a resolver.
+func RegisterProtocol(scheme string, resv Resolver) {
```

Comment:
Ultra-minor - I'd call it just `Register` (as I did), or rename `schema` to `proto` as well... But really, really minor.

Reply:
was following https://golang.org/pkg/net/http/#Transport.RegisterProtocol